python/pyarrow/tests/test_dataset.py

```diff
@@ -905,13 +896,12 @@ def test_fragments_parquet_ensure_metadata(tempdir, open_logging_fs):
     assert row_group.num_rows == 2
     assert row_group.statistics is not None
 
-    # pickling preserves row group ids but not statistics
+    # pickling preserves row group ids and statistics
```

Comment:
the test doesn't really test this aspect, though? (the fact that pickling preserved the statistics, as if they were not preserved, we would load them on access) Maybe putting an `with assert_opens([]):` around the asserts after loading the pickle?

Reply:
The fragment is always reopened currently since all properties of `ParquetFileFragment` ensure complete metadata. Ideally the metadata would be serialized while pickling and unpickling would produce a fully loaded parquet fragment, but this needs to wait for a follow up since there is non trivial mismatch between how parquet views schemas (depth first flat indexed) an how arrow views schemas which currently requires an open reader to mediate. Once we disentangle this we can avoid reopening files.